packages/astro/src/content/types-generator.ts

```diff
@@ -419,6 +431,27 @@ async function writeContentFiles({
 				for (const entryKey of Object.keys(collection.entries).sort()) {
 					const dataType = collectionConfig?.schema ? `InferEntrySchema<${collectionKey}>` : 'any';
 					dataTypesStr += `${entryKey}: {\n	id: ${entryKey};\n  collection: ${collectionKey};\n  data: ${dataType}\n};\n`;
+					if (
+						settings.config.experimental.contentCollectionJsonSchema &&
+						collectionConfig?.schema
+					) {
+						try {
+							await fs.promises.writeFile(
+								new URL(`./${collectionKey.replace(/"/g, '')}.json`, collectionSchemasDir),
+								JSON.stringify(
+									zodToJsonSchema(collectionConfig.schema, {
+										name: collectionKey.replace(/"/g, ''),
+										markdownDescription: true,
+										errorMessages: true,
+									}),
+									null,
+									2
+								)
+							);
+						} catch (error) {
+							// new AstroError(error)
```

Comment:
Still needs to be handled!

Reply:
I was thinking about logging a warning using our logger, but it seems that is not available here.

Throwing an error seems a bit severe, as a dev I would expect to continue even if this experimental step fails?

Reply:
I think we should have access to the logger since `settings` is passed in.

Agree that throwing an unhandled error would be too severe here. It seems intentional to gracefully degrade the experience if we run into problems with this step.